packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,291 @@
+  /// background color. Defaults to the translucent
+  /// [CupertinoColors.tertiarySystemFill] iOS color.
+  final Color? backgroundColor;
+
+  /// The [borderRadius] property lets you set the [decoration] property's
+  /// border radius. Defaults to 9 px circular corner radius.
+  // TODO(DanielEdrisian): Must make border radius continuous, see
+  // https://github.com/flutter/flutter/issues/13914.
+  final BorderRadius borderRadius;
+
+  /// The [padding] property lets you set the padding insets for the text and
+  /// placeholder. Defaults to padding that replicates the `UISearchTextField`
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets padding;
+  final Color? itemColor;
+  final double itemSize;
+
+  /// The [prefixInsets] property lets you set the padding insets for the
+  /// suffix. Defaults to padding that replicates the `UISearchTextField` suffix
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets prefixInsets;
+
+  /// The [suffixInsets] property lets you set the padding insets for the
+  /// prefix. Defaults to padding that replicates the `UISearchTextField` prefix
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets suffixInsets;
+
+  /// The [suffixIcon] property sets the suffix widget's icon. Defaults to
+  /// the X-Mark [CupertinoIcons.xmark_circle_fill]. The suffix is customizable
+  /// so that users can override it with other options, like a bookmark icon.
+  final Icon suffixIcon;
+
+  /// The [suffixMode] property dictates when the X-Mark (suffix) should be
+  /// visible, a.k.a. only on when editing, not editing, on always, or on never.
+  /// This defaults to only on when editing.
+  final OverlayVisibilityMode suffixMode;
+
+  /// The [onSuffixTap] property lets you customize the X-Mark (suffix) action.
+  /// This defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+}
+
+class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField> {
+  @override
+  Widget build(BuildContext context) {
+    // The icon size will be scaled by a factor of the accessibility text scale,
+    // to follow the behavior of `UISearchTextField`.
+    final double scaledIconSize =
+        MediaQuery.textScaleFactorOf(context) * widget.itemSize;
+
+    // If decoration was not provided, create a decoration with the provided
+    // background color and border radius (which both have default values).
+    final BoxDecoration decoration = widget.decoration ??
+        BoxDecoration(
+          color: CupertinoDynamicColor.resolve(widget.backgroundColor, context),
```

Comment:
We should do something similar to https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/container.dart#L275 (and document it). Otherwise it might be confusing that a value was passed in via the constructor and ignored.

Reply:
good idea